### PR TITLE
fix allocations coming from `num_elements`

### PR DESCRIPTION
**Pull request by ranocha**
Fixes https://github.com/trixi-framework/Trixi.jl/issues/1435 on Julia v1.9.0-rc3.

### CODE REVIEW

File: src/MeshData.jl [hunk 1]
```diff
@@ -199,7 +199,7 @@ function Base.getproperty(x::MeshData{Dim, <:VertexMappedMesh}, s::Symbol) where
 end
 
 num_elements(md::MeshData) = size(md.x, 2) # number of columns in the "x" coordinate array
-num_elements(md::MeshData{Dim, <:VertexMappedMesh}) where {Dim} = size(md.mesh_type.EToV, 1)
+num_elements(md::MeshData{Dim, <:VertexMappedMesh}) where {Dim} = size(getfield(md, :mesh_type).EToV, 1)
```
Comment (ranocha) [inline src/MeshData.jl:202]: This is the only real change - everything else is just stripping trailing whitespace (done by my editor automatically)

**Comment by ranocha** [comment]
I don't understand what fails on Windows...

**Comment by jlchan** [comment]
@ranocha thanks for catching that! I'm rerunning the failed windows test, but otherwise LGTM.

**Comment by jlchan** [comment]
Closing in favor of https://github.com/jlchan/StartUpDG.jl/pull/121

**Pull request closed without merging**